app/routes/pages.js

```diff
@@ -0,0 +1,22 @@
+import Ember from 'ember';
+
+const { Route } = Ember;
+
+export default Route.extend({
+  titleToken() {
+    return this.l10n.t('Pages');
+  },
+  model(params) {
+    let externalPage;
+    this.modelFor('application').pages.forEach(page => {
+      if (page.get('url') === params.path) {
+        externalPage = page;
+      }
+    });
+    if (externalPage) {
+      return externalPage;
+    } else {
+      this.transitionTo('not-found');
```

Comment:
@sumedh123 please don't transition to the `not-found` route. We don't want the address bar url to change. Just render the `not-found` template for the same route. 

See how it works now https://open-event-frontend.herokuapp.com/sdfdsf/sdfsdf/ds/f/sdfr